src/main/java/net/tridentsdk/server/net/NetClient.java

```diff
@@ -312,5 +314,12 @@ public void disconnect(Chat reason) {
         } else {
             this.channel.close();
         }
+        if (name != null) {
```

Comment:
Player messages should be associated with a disconnecting _player_ instead of a closed _connection_.